### PR TITLE
JBrowse Tool

### PR DESCRIPTION
**Pull request by erasche**
Release early and release often :)

Currently only supports GFF/3, BED files.

TODO:

- BAM
- Wig/BigWig
- Others?
- Nicer help (but it's pretty obvious what's going on, IMO)

This is a tough tool to test (especially with package dependencies...), eventually I'll make a PR to test the processed data files used in JBrowse. (Maybe once galaxyproject/planemo#19 is done)

image

image 1

jenkins: notest

### CODE REVIEW

**Comment by erasche** [comment]
Jenkins: no test

**Comment by erasche** [comment]
Also, ping @shiltemann in case you're interested in using this in your iReport tool.

**Comment by erasche** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://gx.hx42.org//job/IUC-Github/158/
Test FAILed.

File: tools/jbrowse/jbrowse.xml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+--gff3_label "$data_labels"
+
+
+--jbrowse \${JBROWSE_SOURCE_DIR}
+--outdir $default.files_path
+> $default]]></command>
+  <inputs>
+    <param label="Genome" name="positional_1" type="data" format="fasta"/>
+    <repeat name="data_tracks" title="Annotation Track">
+      <param label="GFF3/BED" format="gff,gff3,bed" name="gff3"
```
Comment (bgruening) [inline tools/jbrowse/jbrowse.xml]: Label is misleading. Maybe label="Annotation data" and help="in GFF, GFF3 and BED format."?

File: tools/jbrowse/macros.xml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  <token name="@WRAPPER_VERSION@">1.0</token>
+  <xml name="stdio">
+    <stdio>
+      <exit_code range="1:"/>
+      <exit_code range=":-1"/>
+      <regex match="Error:"/>
+      <regex match="Exception:"/>
+    </stdio>
+  </xml>
+  <token name="@ATTRIBUTION@"><![CDATA[
```
Comment (bgruening) [inline tools/jbrowse/macros.xml]: I think this is never used.

**Comment by bgruening** [comment]
Big :+1: !

File: tools/jbrowse/jbrowse.xml (hunk 1 above)
Comment (erasche) [inline tools/jbrowse/jbrowse.xml]: Ah, yes, good suggestion.

**Comment by erasche** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://gx.hx42.org//job/IUC-Github/160/
Test FAILed.

**Comment by erasche** [comment]
Thanks @bgruening

**Comment by bgruening** [comment]
@erasche 


Let me know if this hits the TS and I will update the docker image.

**Comment by erasche** [comment]
:) @bgruening it's there! https://twitter.com/Eric_Rasche/status/595338101100183552
https://toolshed.g2.bx.psu.edu/repository?repository_id=a3e608531354bfa6

**Comment by bgruening** [comment]
And here is the commit: https://github.com/bgruening/docker-galaxy-genome-annotation/commit/a2d9b3f232780897d7984e2adc336b0ac59f2920

Thanks a lot!

**Pull request merged**